operator/v1/types_ingress.go

```diff
@@ -492,6 +492,15 @@ type ProviderLoadBalancerParameters struct {
 	//
 	// +optional
 	IBM *IBMLoadBalancerParameters `json:"ibm,omitempty"`
+
+	// openstack provides configuration settings that are specific to OpenStack
+	// load balancers.
+	//
+	// If empty, defaults will be applied. See specific openstack fields for
+	// details about their defaults.
+	//
+	// +optional
+	OpenStack *OpenStackLoadBalancerParameters `json:"openstack,omitempty"`
```

Comment:
You can add CEL validation to the parent struct that ensures this field is not specified when the `type` is not `OpenStack`, it would have to be an optional member so something like
```
// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'OpenStack' ? true : !has(self.openstack)",message="openstack is not permitted when type is not OpenStack"
```